os/fscusat/modules/mirror/archlinux/default.nix

```diff
@@ -0,0 +1,59 @@
+{ pkgs, ... }:
+
+let
+  target = "/var/lib/archlinux";
+  lock = "/var/lock/syncrepo.lck";
+  bwlimit = 0;
+  source_url = "rsync://mirror.guillaumea.fr/archlinux/";
+  lastupdate_url = "https://mirror.guillaumea.fr/archlinux/";
+
+  rsync_cmd = { src, dest, excludes ? [], extraArgs ? [] }@args:
+    let
+      cmd = pkgs.writeScript "rsync_cmd-${src}" ''
+        rsync -rlptH --safe-links --delete-delay --delay-updates \
+          --timeout=600 --contimeout=60 --no-motd \
+          ${if pkgs.stty then "-h -v --progress" else "--quiet"} \
+          ${if bwlimit > 0 then "--bwlimit=$bwlimit" else ""} \
+          ${lib.concatMapStrings (s: " --exclude='${s}'") excludes} \
+          ${src} ${dest} ${lib.concatMapStrings (s: " '${s}'") extraArgs}
+      '';
+    in
+      pkgs.stdenv.mkDerivation {
```

Comment:
This looks like a job for [writeShellApplication](https://ryantm.github.io/nixpkgs/builders/trivial-builders/#trivial-builder-writeShellApplication)